quickwit/quickwit-control-plane/src/model/mod.rs

```diff
@@ -54,7 +54,7 @@ use tracing::{info, instrument, warn};
 #[derive(Default, Debug)]
 pub(crate) struct ControlPlaneModel {
     index_uid_table: FnvHashMap<IndexId, IndexUid>,
-    index_table: FnvHashMap<IndexUid, IndexMetadata>,
+    index_source_table: FnvHashMap<IndexUid, HashMap<SourceId, SourceConfig>>,
```

Comment:
- it is more explicit to only store the sources in the state as the rest of the IndexMetadata isn't relevant
- if we kept the entire `IndexMetadata`, we would need to also proxy the update calls through the control plane

Reply:
Ok I'll revert. Don't you think it would be better to have a smaller struct that only contains the data that is really useful to the control plane? It would make it easier to read what the control plane does.